ProductFunctions/ProductFunctions/ProductFunctions.cs

```diff
@@ -4,93 +4,70 @@
 using Microsoft.Azure.WebJobs.Extensions.Http;
 using Microsoft.AspNetCore.Http;
 using Microsoft.Extensions.Logging;
-using Newtonsoft.Json;
 using System.IO;
-using System;
 using System.Web.Http;
+using System;
+using Newtonsoft.Json;
+using System.Collections.Generic;
 
 namespace ProductFunctions
 {
-    public static class ProductFunctions
-    {
-
-        static DummyDB dummyDB = new DummyDB();
-
-        [FunctionName("getAllProducts")]
-        public static async Task<IActionResult> Run(
-            [HttpTrigger(AuthorizationLevel.Function, "get", Route = "products")] HttpRequest req,
-            ILogger log)
-        {
-            log.LogInformation("C# HTTP trigger function processed a request.");
-
-            var existingProducts = JsonConvert.SerializeObject(dummyDB.onDisplayProducts, Formatting.Indented);
-
-            return new OkObjectResult(existingProducts);
-        }
+  public static class ProductFunctions
+  {
 
-        [FunctionName("addProduct")]
-        public static async Task<IActionResult> AddProduct(
-        [HttpTrigger(AuthorizationLevel.Function, "post", Route = "products/add/")] HttpRequest req,
-    ILogger log)
-        {
-            log.LogInformation("C# HTTP trigger function processed a request.");
-            string requestBody = await new StreamReader(req.Body).ReadToEndAsync();
+    static ProductService productService = new();
 
-            try
-            {
-                dynamic data = JsonConvert.DeserializeObject<Product>(requestBody);
-                dummyDB.onDisplayProducts.Add(data);
-            }
-            catch (Exception ex)
-            {
-                return new BadRequestObjectResult("Not a proper product");
-            }
-
-            return new OkObjectResult("[OK] Product added");
-        }
+    [FunctionName("getAllProducts")]
+    public static async Task<IActionResult> GetProducts(
+        [HttpTrigger(AuthorizationLevel.Function, "get", Route = "products")] HttpRequest req,
+        ILogger log)
+    {
+      int pageNumber = Int32.Parse(req.Query["page"]);
+      int pageSize = Int32.Parse(req.Query["pgsize"]);
+      List<Product> newProducts = productService.GET(pageNumber-1, pageSize);
```

Comment:
Rename method name to GetAllProducts, GET is not very specific, same to all other methods in service